lib/travis/cli/encrypt_file.rb

```diff
@@ -14,6 +14,7 @@ class EncryptFile < RepoCommand
       description 'encrypts a file and adds decryption steps to .travis.yml'
       on '-K', '--key KEY', 'encryption key to be used (randomly generated otherwise)'
       on '--iv IV', 'encryption IV to be used (randomly generated otherwise)'
+      on '--env-prefix ENV_PREFIX', 'prefix of the names of the encryption key/IV environment variables (randomly generated otherwise)'
```

Comment:
Hello, there. Thanks for the PR.

I have a couple of questions:

1. What does this do that the existing `--key` and `--iv` flags do not?
1. The code _defines_ the variable, but it doesn't actually _do_ anything. Did I miss something?

Reply:
I'd need to know the *names* of the two variables in advance.
At the moment, what we have before `_key` and `_iv` is really hard to be determined before running the command (their prefix is determined [here](https://github.com/travis-ci/travis.rb/blob/4b30bdfb27d66a1ad58631cd3e4e18601caed189/lib/travis/cli/encrypt_file.rb#L81)).

If we let users specify this `--env-prefix` option, the `env_prefix` variable will be set to the value of it, and
```ruby
@env_prefix ||= "encrypted_#{Digest.hexencode(Digest::SHA1.digest(input_path)[0..5])}"
```
won't do anything (since `env_prefix` is already defined, it won't be assigned the value of what we have after the `=`).

Reply:
@BanzaiMan Is there something unclear with my answer?

Reply:
No, there isn't. I understand the intent now. One thing still a little uneasy is the name. `env_prefix` seems a little vague. Maybe `arg_prefix`, `env_var_prefix` or something like that. Maybe a short flag name would help as well.

Additionally, the help text should indicate how this would interact with `-K`/`--key` and `--iv`.

Reply:
> One thing still a little uneasy is the name. `env_prefix` seems a little vague. Maybe `arg_prefix`, `env_var_prefix` or something like that.

- if users don't specify this new parameter, the name of the two variables containing the IV/key pair will be prefixed by `encrypted_` followed by a SHA-1 hash (for example: `encrypted_abcdef012345_iv` / `encrypted_abcdef012345_key`)
- if users use this new parameter, they'll be able to specify what should be used instead of `encrypted_abcdef012345`.

So, what about `enc_var_prefix` ?

> Maybe a short flag name would help as well.

Well, I think it'll be a rarely used argument, I don't know if it's worth reserving a short flag just for this tiny feature.

> Additionally, the help text should indicate how this would interact with `-K`/`--key` and `--iv`.

Well, there's no interaction with the value of the generated variables, just with their names.
And to me the current text is quite clear about that:
`prefix of the names of the encryption key/IV environment variables (randomly generated otherwise)`
Do you have a better alternative to it?

Reply:
- the `--key` (aka `-K`) sets the **value** of the encryption key
- the `--iv` sets the **value** of the encryption IV
- this `--enc-var-prefix` sets the prefix of the **names** of the two above values

Examples:

- running `travis encrypt-file file.txt`:
  generates two environment variables:
  - `encrypted_abcdef012345_iv` with a value generated automatically
  - `encrypted_abcdef012345_key` with a value generated automatically
- running `travis encrypt-file --enc-var-prefix myprefix file.txt`:
  generates two environment variables:
  - `myprefix_iv` with a value generated automatically
  - `myprefix_key` with a value generated automatically
- running `travis encrypt-file --iv myiv --key mykey file.txt`:
  generates two environment variables:
  - `encrypted_abcdef012345_iv` with the "myiv" value
  - `encrypted_abcdef012345_key` with the "mykey" value
- running `travis encrypt-file --enc-var-prefix myprefix --iv myiv --key mykey file.txt`:
  generates two environment variables:
  - `myprefix_iv` with the "myiv" value
  - `myprefix_key` with the "mykey" value

As you can see, there's no interaction between `--key`/`-K`/`--iv` and `--enc-var-prefix`.

And that's quite clear to me from the description of the options:

- `--key`: encryption key to be used (randomly generated otherwise)
- `--iv`: encryption IV to be used (randomly generated otherwise)
- `--enc-var-prefix`: prefix of the names of the encryption key/IV environment variables (randomly generated otherwise)